### ISSUE TITLE
IntPtr and UIntPtr now implement IFormattable

### ISSUE THREAD

**Issue by tannergooding** [dotnet/docs#19679]
## IntPtr and UIntPtr now implement IFormattable

`IntPtr` and `UIntPtr` are now implementing `IFormattable`, this means that various functions that check for `IFormattable` support may now return different results for these types as they may pass in a format specifier and a culture.

### Version introduced

.NET 5 Preview 4

### Old behavior

Previous, `IntPtr` and `UIntPtr` did not implement `IFormattable`, so functions would check for `IFormattable` and may have fallen back to just calling `ToString()`, which meant that format specifiers and cultures were not respected.

This is impactful for scenarios like interpolated strings or Console.WriteLine, among others.

### New behavior

`IntPtr` and `UIntPtr` now implement `IFormattable`, this means that various functions that check for `IFormattable` support may now return different results for these types as they may pass in a format specifier and a culture.

### Reason for change

`IntPtr` and `UIntPtr` now have language support in C# via the `nint` and `nuint` keywords. The backing types were updated to provide near parity (where possible) with the functionality exposed by other primitive types such as `System.Int32`

### Recommended action

Users can explicitly call the `ToString()` overload that takes no arguments if they want no format specifier or custom culture to be used.

### Category

Core .NET Libraries

### Affected APIs

"Not detectable via API analysis"

---
#### Issue metadata

* Issue type: breaking-change

**Comment by Youssef1313** [comment]
@BillWagner, The new `nint` and `nuint` keywords will need to be documented in [built-in types](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/built-in-types) article. Should I open a separate issue to keep track of these new keywords in C# 9?

**Comment by BillWagner** [comment]
@Youssef1313 

Yes, they do need to be added. I'm waiting for dotnet/csharplang#3731 to be merged before starting those tasks.  

Watching the [C# language feature status page](https://github.com/dotnet/roslyn/blob/master/docs/Language%20Feature%20Status.md) also helps manage when to make these additions.

**Comment by Youssef1313** [comment]
@BillWagner, Great. Thanks for letting me know you track those.
